### DISCUSSION TITLE
Log full stack trace for exceptions

### DISCUSSION THREAD

**Discussion by jerriep** [statiqdev/Discussions#177] [Q&A]
Hi,

It seems as though when Statiq encounters an unhandled exception, it only logs the exception message and not a full exception with the stack trace

![image](https://user-images.githubusercontent.com/1006420/203493549-3e1f63c7-3103-48d0-aad2-12de8c6ebfbb.png)

Does anyone know whether it is possible to configure the logger, so it logs full stack traces for exceptions?

**Comment by daveaglick** [comment]
Does increasing the log level give you what you need? I. e. `-l trace` (note that you'll also get a lot of other information, but it should show the whole stack trace for exceptions too).

**Reply by jerriep** [reply]
It seems as though setting the `-l` flag has no effect. I've tried setting it to `Trace` and to `Debug`, but I still only get informational level messages. I have confirmed that the command line arguments are passed correctly to `Bootstrapper.Factory.CreateDefault(...)`, so I am not sure what is happing. 

Any idea why that flag will be ignored?

**Reply by daveaglick** [reply]
What does your full command look like? I can't replicate increasing the log level having no impact. Do you see additional data and just not a full stack trace, or does the output look exactly the same as without the `-l Debug` switch? One way to tell if it's working is if you see a whole lot of `[DBUG] Cataloging types in assembly` lines right at the start - those are only output with a debug log level.

**Reply by jerriep** [reply]
This is the command line - with some information redacted:

```
Generator.exe deploy -g -l Debug -n --clean-mode Full --serial -i input\\xxx\\common -s GA_MEASUREMENT_ID=xxx -s DEPLOY_TARGET=AZURE_STATIC_WEBSITE -s BLOB_SERVICE_URL=https://xxx.blob.core.windows.net -s Host=xxx.web.core.windows.net -s LinkHideIndexPages=true -s LinksUseHttps=true -s CategoryPageSize=30
```

There are no Debug entries being logged to the console. Also, the `-l Debug` flag has no effect on the stack trace.

**Reply by daveaglick** [reply]
That all looks correct, so I'm a bit stumped at this point why it's not working. I've added a little hint to the initial set of information Statiq prints out in the next release (going out right now), so I'm curious to see what that says and whether the `-l` flag isn't taking effect for you, or whether it is but the debug messages aren't making it to the console.

![image](https://user-images.githubusercontent.com/1020407/209236931-af4c8058-2a77-4d1a-a549-938ef4d723f7.png)

**Reply by jerriep** [reply]
Thanks, Dave. That will be very helpful
